docs/kratos/emails-sms/05_custom-email-templates.mdx

```diff
@@ -0,0 +1,285 @@
+---
+id: custom-email-templates
+title: Customize message templates for recovery and verification emails
+sidebar_label: Email templates
+---
+
+Ory Identity Service (Ory Kratos) comes with built-in templates for all messages sent by the system. You can replace the default
+templates with custom ones that will carry your own branding, visual elements, and communication tone.
+
+The system uses templates in these flows:
+
+1. `recovery` - the system sends these messages when users recover their accounts, for example after forgetting their password.
+   - `recovery.valid` - the system sends these messages when the user completes the recovery flow successfully.
+   - `recovery.invalid` - the system sends these messages when the recovery flow fails.
+2. `verification` - the system sends these messages when users verify their accounts after signing up.
+   - `verification.valid` - the system sends these messages when the user verifies their account successfully.
+   - `verification.invalid` - the system sends these messages when the account verification fails.
+
+## Use custom message templates
+
+You can use custom templates in place of built-in ones. If you don't specify a custom template, the system automatically uses the
+built-in one.
+
+````mdx-code-block
+import CodeBlock from '@theme/CodeBlock'
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs>
+  <TabItem value="cloud" label="Ory Cloud" default>
+
+To use custom templates in Ory Cloud, encode the content template with Base64 and use the encoded string to update the Identity Service config.
+Follow these steps:
+
+1. List all your projects and get the Identity Service config file:
+
+   ```shell
+   ## List all available projects
+   ory list projects
+
+   ## Get config
+   ory get identity-config <project-id> --format yaml > identity-config.yaml
+   ```
+2. Add your Base64-encoded custom message templates to the config:
+
+   ```yaml title="identity-config.yaml"
+   courier:
+     smtp:
+       from_name: MyProject via Ory
+     templates:
+       recovery:
+         invalid:
+           email:
+             body: {
+               "html": "base64://ENCODED_HTML_TEMPLATE",
+               "plaintext": "base64://ENCODED_PLAINTEXT_TEMPLATE"
+             }
+         valid:
+           email:
+             body: {}
+       verification:
+         invalid:
+           email:
+             body: {}
+         valid:
+           email:
+             body: {}
+    ```
+3. Upload the edited config:
+
+   ```shell
+   ory update identity-config <project-id> --file updated_config.yaml
+   ```
+
+:::warning
+
+In Ory Cloud, you can use only Base64-encoded templates.
```

Comment:
I change the message to strengthen the point that b64 URIs are the only way to go. Added an explanation of what happens when you try to provide templates using unsupported methods (fallback to default templates). 

 Reg. `template_override_path` - I describe it only in the "self-hosted section" for a reason. ;) It's not a sustainable approach to say that certain things from self-hosted don't work in Cloud and vice-versa. Instead, the docs draw the line by explicitly saying to what scenario a given piece of instructions applies. In this document, it's done using the tabs. Fun fact - when the user switches one tab to, let's say, "self-hosted", all other tabs on the page switch to self-hosted as well.

Reply:
True, but since we don't offer some of these "features" in Cloud, people might automatically think self-host and cloud should be 1:1 on "features" but the way we configure them might be different.

I would just strengthen on key areas that it is not possible in cloud in the cloud section (can even be a warning or note at the top of the cloud section).

" in Cloud setting `template_override_path` is unsupported as well as setting any remote URL for template `body` and `subject`..."